pkg/liqo-controller-manager/webhooks/shadowpod/spdescription_test.go

```diff
@@ -0,0 +1,250 @@
+		errTest           error
+		shadowPod         *vkv1alpha1.ShadowPod
+		spDescription     *Description
+		spDescriptionTest *Description
+		spNamespacedName  types.NamespacedName
+		fakeNamespace     *corev1.Namespace
+		peeringInfo       *peeringInfo
+		fakeClient        client.Client
+		quota             *corev1.ResourceList
+		containers        []containerResource
+		initContainers    []containerResource
+	)
+
+	BeforeEach(func() {
+
+		fakeNamespace = forgeNamespaceWithClusterID(clusterID)
+
+		fakeClient = fake.NewClientBuilder().WithScheme(scheme).WithObjects(fakeNamespace).Build()
+
+		spValidator = webhook.Admission{Handler: NewValidator(fakeClient, false)}.Handler.(*Validator)
+
+		peeringInfo = createPeeringInfo(*clusterIdentity, *resourceQuota)
+
+		spNamespacedName = types.NamespacedName{Name: testShadowPodName, Namespace: testNamespace}
+
+		Expect(spValidator.InjectDecoder(decoder)).To(Succeed())
+	})
+
+	Describe("Get or Create a ShadowPod Description", func() {
+		JustBeforeEach(func() {
+			containers = nil
+			containers = append(containers, containerResource{cpu: 100, memory: 100})
+			spDescription, err = peeringInfo.getOrCreateShadowPodDescription(ctx, spValidator.client, forgeShadowPodWithResourceLimits(containers, nil))
+		})
+
+		When("The ShadowPod Description does not exist", func() {
+			BeforeEach(func() {
+				spDescription = nil
+				err = nil
+				delete(peeringInfo.shadowPods, testShadowPodName)
+			})
+			It("should return a new one without errors", func() {
+				Expect(spDescription.running).To(BeTrue())
+				Expect(err).To(BeNil())
+			})
+		})
+		When("The ShadowPod Description exists in cache and is running", func() {
+			BeforeEach(func() {
+				spDescriptionTest = createShadowPodDescription(testShadowPodName, testNamespace, testShadowPodUID, *forgeResourceList(100, 100))
+				errTest = fmt.Errorf("ShadowPod %s is already running", spDescriptionTest.namespacedName.Name)
+				spDescription = nil
+				err = nil
+				peeringInfo.shadowPods[spNamespacedName.String()] = spDescriptionTest
+			})
+			It("should return an error", func() {
+				Expect(spDescription).To(BeNil())
+				Expect(err).ToNot(BeNil())
+				Expect(err.Error()).To(Equal(errTest.Error()))
+			})
+		})
+		When("The ShadowPod Description exists in cache and is not running. The Shadow Pod exists in system snapshot", func() {
+			BeforeEach(func() {
+				spDescriptionTest = createShadowPodDescription(testShadowPodName, testNamespace, testShadowPodUID, *forgeResourceList(100, 100))
+				spDescriptionTest.terminate()
+				spDescription = nil
+				errTest = fmt.Errorf("ShadowPod still exists in the system")
+				err = nil
+				peeringInfo.shadowPods[spNamespacedName.String()] = spDescriptionTest
+				containers = nil
+				containers = append(containers, containerResource{cpu: 100, memory: 100})
+				errClient = fakeClient.Create(ctx, forgeShadowPodWithResourceLimits(containers, nil))
+				Expect(errClient).ToNot(HaveOccurred())
```

Comment:
```suggestion
				Expect(fakeClient.Create(ctx, forgeShadowPodWithResourceLimits(containers, nil)).ToNot(HaveOccurred())
```

Keep it simple, and try not to use global variables if not necessary.